src/sensor_interface.c

```diff
@@ -0,0 +1,241 @@
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/__assert.h>
+
+#include <sensor_interface.h>
+
+LOG_MODULE_DECLARE(HouseEnvironmentMonitor);
+K_LIFO_DEFINE(sensor_value_lifo);
+
+/***** SIMULATION SETTINGS *****/
+#define DEFAULT_TEMPERATURE 250
+#define DEFAULT_PRESSURE 100
+// #define DEFAULT_HUMIDITY 99
+static uint8_t sim_temperature = DEFAULT_TEMPERATURE;
+static uint8_t sim_pressure = DEFAULT_PRESSURE;
+// static uint8_t sim_humidity = DEFAULT_HUMIDITY;
+
+/* This function gets the device from the device tree. Note that the device
+ * should have "bosch,bme280" as compatible string.
+ * 
+ * @param None
+ * @return Pointer to the device, or NULL if the device is not found.
+ */
+const struct device *get_bme280_device(void) {
+	const struct device *const dev = DEVICE_DT_GET_ANY(bosch_bme280);
+
+	if (dev == NULL) {
+		/* No such node, or the node does not have status "okay". */
+		LOG_WRN("No BME280 sensor found. Will be using simulator instead.");
+		return NULL;
+	}
+
+	if (!device_is_ready(dev)) {
+		LOG_ERR("Error: Device \"%s\" is not ready. Check the driver initialization logs for errors.",
+		       dev->name);
+		return NULL;
+	}
+
+	LOG_INF("Found BME280 device \"%s\"", dev->name);
+	return dev;
+}
+
+/* Read the I2C BME280 sensor, or simulate the sensor data if the sensor is not
+ * available. 
+ *
+ * @param dev Pointer to the device (struct device).
+ * @return None
+ */
+void update_sensor_data(const struct device *dev) {
+	// Variables to store sensor data.
+	struct sensor_value temp_value, press_value;
```

Comment:
In general, the notation below is used compared to shorter one. 
```suggestion
	struct sensor_value temp = 0;
    struct press_value press = 0;
```

Reply:
I guess it all depends on the coding style used. Which coding style you're mentioning for this one?

Reply:
I mostly refer to BARR-C:2018 and MISRA C:2012 but for the other standards it is usually same.

https://barrgroup.com/sites/default/files/barr_c_coding_standard_2018.pdf

Here is an example. 

```
	char * x, y; // Was y intended to be a pointer? Does the writer knows that y is not a pointer?
```